ext/app/electron/GristApp.ts

```diff
@@ -1,319 +1,116 @@
-    const debugLogPath = (process.env.GRIST_LOG_PATH ||
-      path.join(this.app.getPath('home'), 'grist_debug.log'));
-
-    if (process.env.GRIST_LOG_PATH || fse.existsSync(debugLogPath)) {
-      const output = fse.createWriteStream(debugLogPath, { flags: "a" });
-      output.on('error', (err) => log.error("Failed to open %s: %s", debugLogPath, err));
-      output.on('open', () => {
-        log.info('Logging also to %s', debugLogPath);
-        output.write('\n--- log starting by pid ' + process.pid + ' ---\n');
-
-        const fileTransportOptions = {
-          name: 'debugLog',
-          stream: output,
-          level: 'debug',
-          timestamp: log.timestamp,
-          colorize: true,
-          json: false
-        };
-
-        // TODO: This does not log HTTP requests to the file. For that we may want to use
-        // "express-winston" module, and possibly update winston (we are far behind).
-        log.add(winston.transports.File, fileTransportOptions);
-        winston.add(winston.transports.File, fileTransportOptions);
-      });
-    }    
-  }
-
-  private async onReady() {
-    // APP_HOME_URL is set by loadConfig
-    this.appHost = process.env.APP_HOME_URL as string;
 
     await updateDb();
-
-    this.flexServer = await mergedServerMain(
-      parseInt(process.env["GRIST_PORT"] as string, 10),
-      ['home', 'docs', 'static', 'app'], {
-        loginSystem: getMinimalElectronLoginSystem.bind(null, this.credential, this.authMode),
-      });
-    const serverMethods = this.flexServer.electronServerMethods;
-    // This function is what we'll call now, and also in onInstanceStart. The latter is used on
-    // Windows thanks to makeSingleInstance, and triggered when user clicks another .grist file.
-    // We can only set this callback once we have serverMethods and appHost.
-    this.onStartup = async (optPath?: string) => {
-      log.debug("onStartup %s", optPath);
-      if (optPath) {
-        await this.handleOpen(serverMethods, optPath);
-        return;
-      }
-      const win = this.getLastWindow();
-      if (win) {
-        win.show();
-        return;
+    const port = parseInt(process.env["GRIST_PORT"] as string, 10);
+    const mergedServer = await MergedServer.create(port, ["home", "docs", "static", "app"]);
+    this.flexServer = mergedServer.flexServer;
+    this.docRegistry = await DocRegistry.create(this.flexServer.getHomeDBManager());
+    this.windowManager = new WindowManager(this.flexServer.getGristConfig(),
+      // TODO: Move this Doc ID lookup function somewhere else.
+      async (docIdOrUrlId) => {
+        if (this.docRegistry.lookupById(docIdOrUrlId) === null) {
+          // DocRegistry does not know about this doc ID, so this must be an URL ID.
+          return (await this.flexServer.getHomeDBManager().connection.createQueryBuilder()
+            .select("docs")
+            .from(Document, "docs")
+            .where('docs.url_id = :urlId', {urlId: docIdOrUrlId})
```

Comment:
It's why our check-in document is `https://gristlabs.getgrist.com/doc/check-ins` instead of `https://gristlabs.getgrist.com/doc/aV3ryL0ngR4nd0mD0c1D`.

AFAIK every Grist document has an URL ID, which defaults to the first several characters of the Doc ID. It is editable, but only via the API. When Grist opens a document, it navigates with the URL ID instead of the Doc ID, although the latter could be manually used to access the document as well.